File: Java/src/main/java/Solution001_100/Solution83.java
package Solution001_100;

import tool.ListNode;

/**
 * @author windows
 */
public class Solution83 {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode root = head;
        while(head != null) {
            while(head.next != null && head.next.val == head.val){
                   head.next = head.next.next;
            }
            head = head.next;
        }
        return root;
    }


    public static void main(String[] args) {
        Solution83 solution83 = new Solution83();
    }
}

File: C++/src/Solution1_100/Solution83.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution83.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


//Definition for singly-linked list.
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};


ListNode* deleteDuplicates(ListNode* head) {
    ListNode* res = head;
    while (head != nullptr && head->next != nullptr){
        ListNode* next = head->next;
        while (next != nullptr && head->val == next->val){
            next = next->next;
        }
        head->next = next;
        head = head->next;
    }
    return res;
}
